WS17_18/05_lifecycle_segues/your assignment/hahn/Yatodoa/Yatodoa/Controller/TagTableViewController.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  TagTableViewController.swift
+//  Yatodoa
+//
+//  Created by Christian Hahn on 04.12.17.
+//  Copyright © 2017 Alexander Dobrynin. All rights reserved.
+//
+
+import UIKit
+
+let Tags = ["pork", "chop", "pastrami", "ball", "tip", "meatloaf", "ribeye", "tenderloin", "hamburger", "biltong", "bacon", "jowl", "strip", "steak", "kielbasa"].map { string in
+    Tag(label: string)
+}
+
+
+enum TagType {
+    case created
+}
+
+extension TagType {
+    
+    init(tag: Tag) {
+        self = .created
+    }
+    
+    var headerTitle: String {
+        switch self {
+            case .created: return "Tags"
+        }
+    }
+    
+    static func < (lhs: TagType, rhs: TagType) -> Bool {
+        return lhs.hashValue < rhs.hashValue
+    }
+}
+
+protocol TagTableViewControllerDelegate {
+    func ttvc(_ vc: TagTableViewController, selectedTag tag: Tag?)
+}
+
+class TagTableViewController: UITableViewController {
+    
+    var tags: [Tag] = [] {
+        didSet {
+            print(tags)
+            
+            tagsSections = tags.groupBy(key: { tag in
+                return TagType(tag: tag)
+            }).sorted { (l, r) -> Bool in
+                return l.key < r.key
+            }
+        }
+    }
+    
+    var tagsSections: [(key: TagType, value: [Tag])] = []
+    
+    var delegate: TagTableViewControllerDelegate?
+    
+    var selectedTag: String?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.tags.append(contentsOf: Tags)
+    }
+    
+    private func populate(_ n: Int, tagNames: [String]) {
+        (0..<n).forEach { _ in // populate data
+            let randomIndex = Int(arc4random_uniform(UInt32(tagNames.count)))
+            tags.append(Tag(label: tagNames[randomIndex]))
+        }
+    }
+    
+    // MARK: - Table view data source
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return tagsSections.count
```

Comment:
das ist ein typisches problem bei dir: du musst dich auf ein "model" einigen, welches du für den gesamten controller verwendest. du mixt nämlich das normale `tags` Array mit dem `tagSections`. ich verstehe, dass das hier gerade ein Überbleibsel ist, allerdings habe ich das schon in den assignments davor gesehen. 

tipp: verwende einfach nur ein model. ich habe bei meiner demo zwei models verwendet, um den unterschied zwischen "data model" und "tableView model (das mit den sections)" zu verdeutlichen. vielleicht war das keine gute Idee, weil es zu Verwirrung führt, wann man welches model verwenden sollte. zudem muss man diese beide in sync halten, was den aufwand erhöht